src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1419,9 +1416,11 @@ internal static bool IsAttributeDefined(RuntimeModule decoratedModule, int decor
 
                         cNamedArgs = Marshal.ReadInt16(blobStart);
                         blobStart = (IntPtr)((byte*)blobStart + 2); // skip namedArgs count
-#if BIGENDIAN
-                        cNamedArgs = ((cNamedArgs & 0xff00) >> 8) | ((cNamedArgs & 0x00ff) << 8);
-#endif
+
+                        if (!BitConverter.IsLittleEndian)
+                        {
+                            cNamedArgs = ((cNamedArgs & 0xff00) >> 8) | ((cNamedArgs & 0x00ff) << 8);
```

Comment:
Is there a BitOps for this code?

Reply:
This just looks like byte swap. Can we not use `BinaryPrimitives.ReverseEndianness`?

Reply:
it looks like a reverse but it also cuts `cNamedArgs`, is not it?
```csharp
int cNamedArgs = 0x1ABCDEF8;
Console.WriteLine((((cNamedArgs & 0xff00) >> 8) | ((cNamedArgs & 0x00ff) << 8)).ToString("X"));
Console.WriteLine(BinaryPrimitives.ReverseEndianness(cNamedArgs).ToString("X"));
```

prints:
```
F8DE
F8DEBC1A
```

Reply:
RIght, but isn't that just `BinaryPrimitives.ReverseEndianness((ushort)(cNamedArgs))` (this is, we are taking the lower word, swapping the two halves, and returning that reuslt)?